src/codeCache.h

```diff
@@ -100,13 +100,38 @@ class CodeBlob {
 
 class FrameDesc;
 
+class CodeCache;
+
+class UnloadProtection {
+  private:
+    CodeCache* _protected_cc;
+    void* _lib_handle;
+
+  public:
+    UnloadProtection(CodeCache *cc);
+    UnloadProtection(UnloadProtection&& other) {
+        _protected_cc = other._protected_cc;
+        _lib_handle = other._lib_handle;
+        other._lib_handle = nullptr;
+    }
+
+    ~UnloadProtection();
+
+    UnloadProtection& operator=(const UnloadProtection& other) = delete;
+
+    void patchImport(ImportId id, void* hook_func) const;
+    bool isValid() const { return _protected_cc != nullptr; }
+};
+
 class CodeCache {
   private:
     char* _name;
     short _lib_index;
     const void* _min_address;
     const void* _max_address;
     const char* _text_base;
+    // TODO: Cleanup usages of lib.image_base
```

Comment:
Is this leftover?

Reply:
This is something we should do, but could pollute this PR. Basically there are some cases where we pass both `cc` and `lib.image_base` to some function, which we could now simplify to passing only `cc`. What's your preference?

Reply:
Since `image_base` is added to the `CodeCache` in the current PR, we could include this instead of keeping a TODO comment.

Reply:
15eeade44555c0ef6381f9ed4bfe62f651a83a87